### PR TITLE
Update docs for v3

### LINKED ISSUE #410: Update docs
1. https://github.com/bakdata/kpops/blob/v3/docs/docs/resources/pipeline-config/config.yaml#L19-L21

```diff
- default_output_topic_name: ${pipeline.name}-${component_name}
+ default_output_topic_name: ${pipeline.name}-${component.name}
  # Configures the value for the variable ${error_topic_name}
- default_error_topic_name: ${pipeline.name}-${component_name}-error
+ default_error_topic_name: ${pipeline.name}-${component.name}-error
```

2. https://github.com/bakdata/kpops/blob/v3/docs/docs/user/core-concepts/variables/substitution.md?plain=1#L29-L30

```diff
-   `error_topic_name` is an alias for `topic_name_config.default_error_topic_name`  
+   `error_topic_name` is an alias for `config.topic_name_config.default_error_topic_name`  
-   `output_topic_name` is an alias for `topic_name_config.default_output_topic_name`
+   `output_topic_name` is an alias for `config.topic_name_config.default_output_topic_name`
```

3. https://github.com/bakdata/kpops/blob/v3/docs/docs/user/core-concepts/variables/substitution.md?plain=1#L24

Mention the prefix!

### PR DESCRIPTION
**Pull request by sujuka99**
Closes #410

### CODE REVIEW

**Review by irux — APPROVED** [review]
LGTM!

File: docs/docs/user/core-concepts/variables/substitution.md [hunk 1]
```diff
@@ -23,11 +23,13 @@ All of them are prefixed with `component.` and follow the following form: `compo
...
 
+All such variables are prefixed with `config.` and are of the same form as the [component-specific variables](#component-specific-variables).
+
 <!-- dprint-ignore-start -->
 
 !!! info Aliases
-    `error_topic_name` is an alias for `topic_name_config.default_error_topic_name`  
-    `output_topic_name` is an alias for `topic_name_config.default_output_topic_name`
+    `error_topic_name` is an alias for `config.topic_name_config.default_error_topic_name`  
+    `output_topic_name` is an alias for `config.topic_name_config.default_output_topic_name`
```
Comment (disrupted) [inline docs/docs/user/core-concepts/variables/substitution.md:32]: in the future, we could rename those to something and deprecate the hardcoded aliases, e.g.
`${config.kafka.error_topic_name}`
`${config.kafka.output_topic_name}`

File: docs/docs/resources/pipeline-config/config.yaml
Comment (disrupted) [inline docs/docs/resources/pipeline-config/config.yaml:1]: why aren't we generating this? 😄

**Review by disrupted — APPROVED** [review]

File: docs/docs/resources/pipeline-config/config.yaml
Comment (sujuka99) [reply docs/docs/resources/pipeline-config/config.yaml:1]: We definitely should be!

The `kpops init` PR is going to solve that problem.

**Pull request merged**